src/contracts/cloud/ComputeRegistry.sol

```diff
@@ -0,0 +1,200 @@
+
+import "../mixins/PermissionControllerMixin.sol";
+import "../mixins/SignatureUtilsMixin.sol";
+import "./ComputeRegistryStorage.sol";
+
+/**
+ * @title ComputeRegistry
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @notice This contract handles permissionless (de)registration of AVS operator sets to the EigenCompute Operator.
+ * It enables AVSs to easily access managed operator infrastructure as part of EigenCloud for quick bootstrapping.
+ */
+contract ComputeRegistry is
+    Initializable,
+    OwnableUpgradeable,
+    ComputeRegistryStorage,
+    PermissionControllerMixin,
+    SignatureUtilsMixin
+{
+    using OperatorSetLib for OperatorSet;
+
+    /**
+     *
+     *                         MODIFIERS
+     *
+     */
+
+    /**
+     * @dev Validates that the operator set exists in the AllocationManager
+     * @param operatorSet The operator set to validate
+     */
+    modifier isValidOperatorSet(
+        OperatorSet memory operatorSet
+    ) {
+        require(ALLOCATION_MANAGER.isOperatorSet(operatorSet), InvalidOperatorSet());
+        _;
+    }
+
+    /**
+     *
+     *                         INITIALIZING FUNCTIONS
+     *
+     */
+
+    /**
+     * @dev Initializes the contract with immutable values
+     * @param _releaseManager The ReleaseManager contract address
+     * @param _allocationManager The AllocationManager contract address
+     * @param _permissionController The PermissionController contract address
+     * @param _version The semantic version of the contract
+     */
+    constructor(
+        IReleaseManager _releaseManager,
+        IAllocationManager _allocationManager,
+        IPermissionController _permissionController,
+        string memory _version
+    )
+        ComputeRegistryStorage(_releaseManager, _allocationManager)
+        PermissionControllerMixin(_permissionController)
+        SignatureUtilsMixin(_version)
+    {
+        _disableInitializers();
+    }
+
+    /**
+     * @notice Initializes the contract
+     * @param _owner The owner of the contract
+     * @param _tosHash The hash of the Terms of Service that AVS operators must sign
+     */
+    function initialize(address _owner, bytes32 _tosHash) external initializer {
+        __Ownable_init();
+        transferOwnership(_owner);
+        _setTosHash(_tosHash);
+    }
+
+    /**
+     *
+     *                         EXTERNAL FUNCTIONS
+     *
+     */
+
+    /**
+     * @inheritdoc IComputeRegistry
+     */
+    function registerForCompute(
+        OperatorSet memory operatorSet,
+        bytes memory signature
+    ) external checkCanCall(operatorSet.avs) isValidOperatorSet(operatorSet) {
+        // Check if there is at least one release for the operator set
+        // The ReleaseManager will revert with `NoReleases()` if there are no releases for the operator set
+        RELEASE_MANAGER.getLatestRelease(operatorSet);
```

Comment:
Thinking out loud if we need a `try-catch` here. I personally think we don't and this is cleaner with better tradeoffs.